save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/DebugInfoStorage.kt

```diff
@@ -27,34 +30,30 @@ class DebugInfoStorage(
     configProperties: ConfigProperties,
     private val objectMapper: ObjectMapper,
 ) :
-    AbstractFileBasedStorage<Pair<Long, TestResultLocation>>(Path.of(configProperties.fileStorage.location) / "debugInfo") {
+    AbstractFileBasedStorage<DebugInfoStorageKey>(Path.of(configProperties.fileStorage.location) / "debugInfo") {
     /**
      * @param rootDir
      * @param pathToContent
      * @return true if path endsWith [SUFFIX_FILE_NAME]
      */
-    override fun isKey(rootDir: Path, pathToContent: Path): Boolean = pathToContent.name.endsWith(SUFFIX_FILE_NAME)
+    override fun isKey(rootDir: Path, pathToContent: Path): Boolean =
+            pathToContent.name.endsWith(SUFFIX_FILE_NAME) && pathToContent.countPartsTill(rootDir) == PATH_PARTS_COUNT
 
     /**
      * @param rootDir
      * @param pathToContent
      * @return [Pair] of executionId and [TestResultLocation] object is built by [Path]
      */
     @Suppress("MAGIC_NUMBER", "MagicNumber")
-    override fun buildKey(rootDir: Path, pathToContent: Path): Pair<Long, TestResultLocation> {
-        val folderNames = generateSequence(pathToContent, Path::getParent)
-            .takeWhile { it != rootDir }
-            .map { it.name }
-            .toList()
-        require(folderNames.size == 5) {
-            "Invalid path to debugInfo: $pathToContent"
-        }
-        val testName = folderNames[0].dropLast(SUFFIX_FILE_NAME.length)
-        val testLocation = folderNames[1]
-        val testSuiteName = folderNames[2]
-        val pluginName = folderNames[3]
-        val executionId = folderNames[4].toLong()
-        return Pair(
+    override fun buildKey(rootDir: Path, pathToContent: Path): DebugInfoStorageKey {
+        val pathNames = pathToContent.pathNamesTill(rootDir)
```

Comment:
Generally I meant this conversation from `pathNames` to set of fields.
I also see, that we use similar logic in other places, i.e. for `TestSuitesSourceSnapshotKey`
May be we can extract this logic into some method at least?
And not write `pathNames[1], pathNames[2], pathNames[3]...` everywhere?

Reply:
this logic of `partNames` depends on storage implementation: it depends on how we put files in file system.
it can be moved to `*Key` implementation as one of constructor (which takes list of string).  but we just move this logic to another place.

As another idea: we can create `*.metadata` file together with uploaded file. It also allows to store additional info without adding it to path. It fits to S3 which has metadata for objects. But I would like to leave this to next phase (I will need to add commit message to snapshots of test suites source), since it will require prepare transition steps

For now, I can re-write it using destructing declaration:
```suggestion
        val (testName, testLocation, testSuiteName, pluginName, executionId) = pathNames
        return DebugInfoStorageKey(
            executionId.toLong(),
            TestResultLocation(testSuiteName, pluginName, testLocation, testName.dropLast(SUFFIX_FILE_NAME.length))
        )
```

Reply:
Yeah, destructing declaration looks good for now